examples/diffdock/sanitize_ligand.py

```diff
@@ -0,0 +1,122 @@
+# pylint: disable=E0401,E1101,I1101
+"""Filter molecules with kekulization errors.
+Handle molecules with sanitization errors by assign formal charge based on valence."""
+# https://depth-first.com/articles/2020/02/10/a-comprehensive-treatment-of-aromaticity-in-the-smiles-language/
+import argparse
+from pathlib import Path
+from typing import Dict, Tuple
+
+import rdkit
+from rdkit import Chem
+from rdkit.Chem import AllChem
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--input_small_mol_ligand', type=str, help='Ligand file')
+args = parser.parse_args()
+input_small_mol_ligand = Path(args.input_small_mol_ligand).name
+
+
+def adjust_formal_charges(molecule: Chem.SDMolSupplier) -> Chem.SDMolSupplier:
+    """Sometimes input structures do not have correct formal charges corresponding
+    to bond order topology. So choose to trust bond orders assigned and generate formal
+    charges based on that.
+    Explicit valence determined what the formal charge should be from dictionary of valence
+    to formal charge for that atomic number. Special case if atom == carbon or nitrogen
+    and if neighbors contain nitrogen, oyxgen or sulfur (polarizable atoms) then if carbon
+    and explicit valence only 3, give formal charge of +1 (more stable then -1 case).
+
+    Args:
+        molecule (Chem.SDMolSupplier): The rdkit molecule object
+
+    Returns:
+        Chem.SDMolSupplier: Molecule object with adjusted formal charges
+    """
+    # 1=H, 5=B, 6=C, 7=N, 8=O, 15=P, 16=S, 17=Cl, 9=F, 35=Br, 53=I
+    atomicnumtoformalchg: Dict[int, Dict[int, int]] = {1: {2: 1}, 5: {4: 1}, 6: {3: -1}, 7: {2: -1, 4: 1},
+                                                       8: {1: -1, 3: 1}, 15: {4: 1}, 16: {1: -1, 3: 1, 5: -1},
+                                                       17: {0: -1, 4: 3}, 9: {0: -1}, 35: {0: -1}, 53: {0: -1}}
+    for atom in molecule.GetAtoms():
+        atomnum = atom.GetAtomicNum()
+        val = atom.GetExplicitValence()
+        if atomicnumtoformalchg.get(atomnum) is None:
+            continue
+        valtochg = atomicnumtoformalchg[atomnum]
+        chg = valtochg.get(val, 0)
+        # special case of polar neighbors surrounding carbon or nitrogen
+        # https://docs.eyesopen.com/toolkits/cpp/oechemtk/valence.html
+        # https://www.masterorganicchemistry.com/2011/03/11/3-factors-that-stabilize-carbocations/
```

Comment:
This is a rather long article. If there is a specific sub-section that is relevant to this code, then please cite the specific sub-section.

Reply:
I see you have added the text "Factors That Stabilize Carbocations – Adjacent Lone Pairs" but the addition of that text doesn't necessarily indicate to the reader that you are attempting to cite a specific section. The article has numbered sections, so you should **cite the section number**: For example:
`See Section 6: Factors That Stabilize Carbocations – Adjacent Lone Pairs`